cmd/spire-server/cli/run/run.go

```diff
@@ -525,6 +549,67 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
 	return sc, nil
 }
 
+func parseBundleEndpointProfile(config federatesWithConfig) (trustDomainConfig *bundleClient.TrustDomainConfig, err error) {
+	var data bytes.Buffer
+	if err := printer.DefaultConfig.Fprint(&data, config.BundleEndpointProfile); err != nil {
+		return nil, err
+	}
+	configString := data.String()
+	profileConfig := new(bundleEndpointProfileConfig)
+	if err := hcl.Decode(profileConfig, configString); err != nil {
+		return nil, fmt.Errorf("failed to decode configuration: %w", err)
+	}
+
+	var endpointProfile bundleClient.EndpointProfileInfo
+	switch {
+	case profileConfig.HTTPSWeb != nil:
+		endpointProfile = bundleClient.HTTPSWebProfile{}
+	case profileConfig.HTTPSSPIFFE != nil:
+		spiffeID, err := spiffeid.FromString(profileConfig.HTTPSSPIFFE.EndpointSPIFFEID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get endpoint SPIFFE ID: %w", err)
+		}
+		endpointProfile = bundleClient.HTTPSSPIFFEProfile{EndpointSPIFFEID: spiffeID}
+	default:
+		return nil, errors.New("no bundle endpoint profile defined")
```

Comment:
Related to the comment above ... I think we may need to pass TD name into this function so that we can get error messages that are more useful. It would be nice to know which TD is misconfigured when we log this error

Reply:
The caller to this function already includes the trust domain as part of the error description. This error is embedded in the error returned by the caller. E.g. the error will say `error parsing federation relationship for trust domain "domain1.test": no bundle endpoint profile defined`